pkg/resource/podtemplatespec.go

```diff
@@ -367,27 +367,43 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, proxyLister configl
 		resources = *cr.Spec.Resources
 	}
 
-	var affinity *corev1.Affinity
-	if cr.Spec.Affinity != nil {
-		affinity = cr.Spec.Affinity
-	} else {
+	// if user has provided an affinity through config spec we use it here, if not
+	// then we fallback to a preferred affinity configuration. we only require a
+	// certain affinity during schedule if the number of replicas is defined to two.
+	affinity := cr.Spec.Affinity
+	if affinity == nil {
 		affinity = &corev1.Affinity{
 			PodAntiAffinity: &corev1.PodAntiAffinity{
 				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
 					{
 						Weight: 100,
 						PodAffinityTerm: corev1.PodAffinityTerm{
-							//TODO godoc for this field says it cannot be empty, but the doc at
-							// https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#an-example-of-a-pod-that-uses-pod-affinity
-							// talks about using an empty topologyKey with anti-affinity as signifying "all topologies",
-							// That said, the standard kubernetes.io/hostname has appeared sufficient in testing on AWS clusters with 3 worker nodes
 							TopologyKey: "kubernetes.io/hostname",
-							Namespaces:  []string{defaults.ImageRegistryOperatorNamespace},
+							Namespaces: []string{
+								defaults.ImageRegistryOperatorNamespace,
+							},
 						},
 					},
 				},
 			},
 		}
+		if cr.Spec.Replicas == 2 {
+			affinity = &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							TopologyKey: "kubernetes.io/hostname",
+							Namespaces: []string{
+								defaults.ImageRegistryOperatorNamespace,
+							},
+							LabelSelector: &metav1.LabelSelector{
```

Comment:
Why do you need the label selector only when you require anti-affinity? Shouldn't the PodAffinityTerm be the same in both cases?

Reply:
I have played a little bit with these Affinity settings and the outcome is that we also need to include this LabelSelector on both. Adjusted accordingly. I am not even sure what was being requested without the LabelSelector in the first place. PTAL